lib/idl.ml

```diff
@@ -241,11 +242,19 @@ end
 exception NoDescription
 
 type rpcfn = Rpc.call -> Rpc.response
-type server_implementation = (string, rpcfn) Hashtbl.t
+type server_implementation = (string, rpcfn option) Hashtbl.t
 
 let server hashtbl call =
+  let unbound_impls = Hashtbl.fold (fun key fn acc ->
+      if fn = None then key::acc else acc
+    ) hashtbl [] in
+  if unbound_impls <> [] then
+    raise (UnboundImplementation unbound_impls);
+
   let fn = try Hashtbl.find hashtbl call.Rpc.name with Not_found -> raise (UnknownMethod call.Rpc.name) in
-  fn call
+  match fn with
+  | Some fn -> fn call
+  | None -> failwith "Unboud implementation. This is impossible"
```

Comment:
I wonder whether we should eliminate this - perhaps your fold could make a new `(string,rpcfn) Hashtbl.t` while checking for unbound impls?

Reply:
Sounds good!